front/lib/models/assistant/actions/process.ts

```diff
@@ -170,7 +191,19 @@ AgentProcessAction.init(
     },
     schema: {
       type: DataTypes.JSONB,
-      allowNull: false,
+      allowNull: true,
+      defaultValue: null,
+    },
+    jsonSchema: {
+      type: DataTypes.JSONB,
+      allowNull: true,
+      get() {
+        const rawSchema = this.getDataValue("schema");
+        const jsonSchema = this.getDataValue("jsonSchema");
+        return jsonSchema || rawSchema
+          ? renderSchemaPropertiesAsJSONSchema(rawSchema || [])
+          : null;
```

Comment:
this is unclear - shouldn't it be 
```suggestion
        return jsonSchema || (rawSchema
          ? renderSchemaPropertiesAsJSONSchema(rawSchema || [])
          : null);
```
?